### DISCUSSION TITLE
Does starship need to be installed on remote machine to work?

### DISCUSSION THREAD

**Discussion by ryanmsnyder** [starship/starship#5037] [Q&A]
The `hostname` module has an `ssh_only` option, which makes me believe that Starship will continue to work when connected to a remote machine via SSH without installing Starship on the remote machine. Is that correct?

**Comment by jankatins** [comment]
No: it has to be installed on all systems which run a shell, as it is integrated into a the process which displays the prompt (=the remote shell). It also doesn't work if you have starship integrated with zsh and then open a bash shell, you would need to integrate it with both.

The [ssh flag comes from seeing a `SSH_CONNECTION` env variable](https://github.com/starship/starship/blob/master/src/modules/hostname.rs#L8-L20), which is set on the remote side (probably by the sshd)

**Reply by ryanmsnyder** [reply]
Thank you. It makes sense that it would need to be installed on the remote host as well. However, the `ssh_only` option on the `hostname` module still confuses me. If you set `ssh_only = false` in `starship.toml` locally, `hostname` will always show in your local shell (that makes sense to me). But what is the point of being able to set it to `true`? If it's set to `true` and you ssh into a remote machine that _doesn't_ have starship installed, nothing will happen. If it's set to `true` and you ssh into a remote machine that _does_ have starship installed, the remote shell will just use the remote machine's `starship.toml` and disregard the local config altogether. The `ssh_only` flag seems useless (though I'm obviously misunderstanding something).

**Reply by jankatins** [reply]
I usually don't ssh into any of my "dev" machine anymore, but I did for some time. During that time, I did sync relevant dotfiles to that machine and so had the shell config... I think at that time it was some zsh function which did a bunch of `scp`s...
